src/commands/captureDiagnostics.ts

```diff
@@ -0,0 +1,209 @@
+        const result = await vscode.window.showInformationMessage(
+            `Saved diagnostic logs to ${diagnosticsDir}`,
+            showInFinderButton,
+            copyPath
+        );
+        if (result === copyPath) {
+            vscode.env.clipboard.writeText(diagnosticsDir);
+        } else if (result === showInFinderButton) {
+            exec(showDirectoryCommand(diagnosticsDir), error => {
+                // Opening the explorer on windows returns an exit code of 1 despite opening successfully.
+                if (error && process.platform !== "win32") {
+                    vscode.window.showErrorMessage(
+                        `Failed to open ${showCommandType()}: ${error.message}`
+                    );
+                }
+            });
+        }
+    } catch (error) {
+        vscode.window.showErrorMessage(`Unable to capture diagnostic logs: ${error}`);
+    }
+}
+
+async function writeLogFile(dir: string, name: string, logs: string) {
+    if (logs.length === 0) {
+        return;
+    }
+    await fs.writeFile(path.join(dir, name), logs);
+}
+
+function extensionLogs(ctx: WorkspaceContext): string {
+    return ctx.outputChannel.logs.join("\n");
+}
+
+function environmentLogs(ctx: WorkspaceContext): string {
+    const environmentOutputChannel = new SwiftOutputChannel("Swift", false);
+    ctx.toolchain.logDiagnostics(environmentOutputChannel);
+    environmentOutputChannel.log("Extension Settings:");
+    environmentOutputChannel.log(
+        JSON.stringify(vscode.workspace.getConfiguration("swift"), null, 2)
+    );
+    return environmentOutputChannel.logs.join("\n");
+}
+
+function diagnosticLogs(): string {
+    const diagnosticToString = (diagnostic: vscode.Diagnostic) => {
+        return `${severityToString(diagnostic.severity)} - ${diagnostic.message} [Ln ${diagnostic.range.start.line}, Col ${diagnostic.range.start.character}]`;
+    };
+
+    return vscode.languages
+        .getDiagnostics()
+        .map(
+            ([uri, diagnostics]) => `${uri}\n\t${diagnostics.map(diagnosticToString).join("\n\t")}`
+        )
+        .join("\n");
+}
+
+function sourceKitLogs(ctx: WorkspaceContext) {
+    return (ctx.languageClientManager.languageClientOutputChannel?.logs ?? []).join("\n");
+}
+
+async function sourcekitDiagnose(ctx: WorkspaceContext, dir: string) {
+    const sourcekitDiagnosticDir = path.join(dir, "sourcekit-lsp");
+    await fs.mkdir(sourcekitDiagnosticDir);
+
+    const toolchainSourceKitLSP = ctx.toolchain.getToolchainExecutable("sourcekit-lsp");
+    const lspConfig = configuration.lsp;
+    const serverPathConfig = lspConfig.serverPath;
+    const serverPath = serverPathConfig.length > 0 ? serverPathConfig : toolchainSourceKitLSP;
+
+    await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+        },
+        async progress => {
+            progress.report({ message: "Capturing Diagnostics..." });
```

Comment:
Should this be something like _Generating SourceKit-LSP Diagnose Bundle_? _Capturing Diagnostics_ sounds to much like the the diagnostics aka live issues to me.

Reply:
I want to keep this message a bit more ambiguous, because if you're not using a Swift 6 toolchain then `sourcekit-lsp diagnose` isn't available. We fall back to recording the captured logs from the sourcekit-lsp output channel, but this isn't really the same thing as the full diagnostics bundle.

Reply:
What do you think about _Diagnosing SourceKit-LSP…_? That should resolve the ambiguity with diagnostics aka. live issues.